frontend/src/components/navigations/top-nav-bar/TopNavBar.jsx

```diff
@@ -33,34 +32,20 @@ function TopNavBar() {
   const baseUrl = getBaseUrl();
   const onBoardUrl = baseUrl + `/${orgName}/onboard`;
   const logout = useLogout();
-  const axiosPrivate = useAxiosPrivate();
+  const [showOnboardBanner, setShowOnboardBanner] = useState(false);
   const { setAlertDetails } = useAlertStore();
   const handleException = useExceptionHandler();
-  const [showOnboardBanner, setShowOnboardBanner] = useState(false);
 
   useEffect(() => {
-    getAdapters();
-  }, []);
+    updateOnBoardBannerStatus();
+  }, [sessionDetails]);
 
-  const getAdapters = () => {
-    const requestOptions = {
-      method: "GET",
-      url: `/api/v1/unstract/${sessionDetails?.orgId}/adapter/`,
-    };
-
-    axiosPrivate(requestOptions)
-      .then((res) => {
-        const data = res?.data;
-        const adapterTypes = [
-          ...new Set(data?.map((obj) => obj?.adapter_type.toLowerCase())),
-        ];
-        if (!onboardCompleted(adapterTypes)) {
-          setShowOnboardBanner(true);
-        }
-      })
-      .catch((err) => {
-        setAlertDetails(handleException(err));
-      });
+  const updateOnBoardBannerStatus = () => {
+    if (onboardCompleted(sessionDetails?.adapters)) {
+      setShowOnboardBanner(false);
+    } else {
+      setShowOnboardBanner(true);
+    }
```

Comment:
```suggestion
    setShowOnboardBanner(!onboardCompleted(sessionDetails?.adapters));
```

Reply:
@mohamed-siddhiq this can be replaced with my suggestion. Pls test it if u making this change.

Reply:
@vishnuszipstack Addressed your comment.